### PR TITLE
Fixes changelings being charged double for reviving stasis

### LINKED ISSUE #82394: Changeling Reviving Stasis costs 30 chemicals instead of 15
## Issue Summary

Changeling reviving stasis says in its tooltip that it costs 15 chemicals to use. I used it the other round and it cost 15 to go down and 15 to get back up. I found that odd, because why would it say 15 instead of 30? So, I took a look at the code and did some testing: there is actual explicit code for setting the cost to 0 to revive from the stasis, however there is an early signal being sent in the middle of the revival that causes the cost to be reset to default unintentionally early. I don't know how signals work, but I want to submit this bug report so I can reference it in the PR fix when I figure out signals.

## Reproduction:

Go into reviving stasis. See that you get charged 15 chems to go down, and 15 chems to get up. Take a look at the code in fakedeath.dm

### PR DESCRIPTION
**Pull request by Metekillot**
## About The Pull Request

Made changes to fakedeath.dm for changelings to stop them from being charged double the chemicals for reviving stasis. Also removed a redundant call to disable_revive, as it's already handled by signals.
## Why It's Good For The Game

Fixes #82394 
## Changelog

Changelings should now be correctly charged 15 chemicals for reviving stasis, instead of 30.
:cl: Bisar
fix: Changelings should no longer be charged double for reviving stasis, instead now only costing 15 chemicals total.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/changeling/powers/fakedeath.dm [hunk 1]
```diff
@@ -19,7 +19,8 @@
 	..()
 	if(revive_ready)
 		INVOKE_ASYNC(src, PROC_REF(revive), user)
-		disable_revive(user) // this should be already called via signal, but just incase something wacky happens
+		//removed the double call to disable_revive, it's already called in revive() via cure_fakedeath
+		//calling to the REMOVE_TRAIT signal
```
Comment (san7890) [inline code/modules/antagonists/changeling/powers/fakedeath.dm:23]: historical comment is not useful here, kindly remove it. anyone with a stake in the code can see revive() do it by looking at the invoke_async at the line above (and this comment will likely degrade quickly should the actual code change and no one fix the comment)

File: code/modules/antagonists/changeling/powers/fakedeath.dm [hunk 2]
```diff
@@ -51,7 +52,13 @@
...
 /datum/action/changeling/fakedeath/proc/disable_revive(mob/living/changeling)
 	if(revive_ready)
-		chemical_cost = 15
+		// added a spawn(1) before we return chemical_cost to 15,
+		// as this verb is callstacked up to sting_action, whose return value is checked
+		// by try_to_sting. try_to_sting doesn't deduct chemicals until AFTER the whole
+		// revival is resolved, so we're delaying it with the spawn to stop the double
+		// chemical deduction. see line 74
+		spawn(1)
+			chemical_cost = 15
```
Comment (san7890) [inline code/modules/antagonists/changeling/powers/fakedeath.dm:61]: please don't invoke `spawn(1)` directly in code like this, we consider these sections of the code to be more "user-facing" (as in it's reasonably expected for any new contributor to start crawling in here and toying with stuff and being utterly confused despite the explanation). if you really need to having a blocking sleep or something I would rather you invoke `sleep(0.1 SECONDS)` instead, but i feel like there should be a better answer to this anyways.

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/changeling/powers/fakedeath.dm (hunk 2 above)
Comment (san7890) [reply code/modules/antagonists/changeling/powers/fakedeath.dm:61]:
```txt
	// This is one of very few spawn()s left in our codebase
	// Yes it needs to be like this
	// No, you're not allowed to use spawn() yourself, unless you're making something that needs to work outside timers
	// Don't be a dumbass, I will always be watching
```

pretty much the only uses of spawn() in the code nowadays are stuff that is placed direclty on `World` or `ASYNC` stuff. you will give poor lemoninthedark a big frowny face if this makes it into production code and i don't know how many more times his muscles can make that face before it gets permanently fused into that position.

Comment (Metekillot) [reply code/modules/antagonists/changeling/powers/fakedeath.dm:61]: i'll need to refactor the code if i'm not able to use the spawn() there. the code to reset the chemical cost is called right in the middle of the revive() proc. i'll make another PR, as I'll have to go at this from the ground up in that case.

Comment (Metekillot) [reply code/modules/antagonists/changeling/powers/fakedeath.dm:61]: in any case i accidentally pushed another entirely separate PR into this one so closing it is probably best, haha.

Comment (san7890) [reply code/modules/antagonists/changeling/powers/fakedeath.dm]: Yeah I think a deep cleaning of how this works is warranted if that's the case. Hrm

**Pull request closed without merging**